src/pysdmx/io/csv/sdmx10/reader/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""SDMX 1.0 CSV reader module."""
+
+from io import StringIO
+from typing import Dict
+
+import pandas as pd
+
+from pysdmx.model.dataset import Dataset
+
+
+def __generate_dataset_from_sdmx_csv(data: pd.DataFrame) -> Dataset:
+    # Extract Structure type and structure id
+
+    # For SDMX-CSV version 1, use 'DATAFLOW' column as the structure id
+    structure_id = data["DATAFLOW"].iloc[0]
+    # Structure type will be "dataflow" in both versions
+    structure_type = "dataflow"
+    # Drop 'DATAFLOW' column from DataFrame
+    df_csv = data.drop(["DATAFLOW"], axis=1)
+
+    # Return a Dataset object with the extracted information
+    return Dataset(
+        unique_id=structure_id,
+        structure_type=structure_type,
+        data=df_csv,
+        attached_attributes={},
+    )
+
+
+def read(infile: str) -> Dict[str, Dataset]:
+    """Reads csv file and returns a payload dictionary.
+
+    Args:
+        infile: Path to file, str.
+
+    Returns:
+        payload: dict.
+
+    Raises:
+        Exception: If it is an invalid CSV file.
+    """
+    # Get Dataframe from CSV file
+    df_csv = pd.read_csv(StringIO(infile))
+    # Drop empty columns
+    df_csv = df_csv.dropna(axis=1, how="all")
+
+    # Determine SDMX-CSV version based on column names
+    if "DATAFLOW" not in df_csv.columns:
+        # Raise an exception if the CSV file is not in SDMX-CSV format
+        raise Exception("Invalid CSV file, only SDMX-CSV is allowed")
```

Comment:
I think we should use pysdmx errors instead (cf. pysdmx/errors.py)?

Reply:
Reviewing exceptions and changing them to ClientError